packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx

```diff
@@ -22,13 +37,40 @@ export const DashboardExtension = ({
   const location = useLocation();
   const navLink = useMemo(() => decodeURIComponent(last(location.pathname.split('/'))), [location.pathname]);
 
+  type MenuTitle = keyof typeof MenuIcons;
+
+  const MenuIcons = {
+    'Patient Summary': ReportIcon,
+    'Vitals & Biometrics': ActivityIcon,
+    Medications: MedicationIcon,
+    Orders: ShoppingCartIcon,
+    Results: ChartAverageIcon,
+    Visits: CalendarHeatMapIcon,
+    Allergies: WarningIcon,
+    Conditions: ListCheckedIcon,
+    Immunizations: SyringeIcon,
+    Attachments: DocumentAttachmentIcon,
+    Programs: ProgramsIcon,
+    Appointments: EventScheduleIcon,
+  } as const;
+
+  const menuIcon = (title: MenuTitle) => {
+    const Icon = MenuIcons[title];
+    return Icon ? <Icon className={styles.icon} /> : null;
+  };
```

Comment:
There's no good reason for these to be located inside this component. They would be better located outside of it.

Reply:
Honestly, though, I think this might not be the right way to design things. This works, but only for providing icons for known menu titles, which is bad for extensibility. Better would be to update the `DashboardExtensionProps` to take a string of the name icon and just render it like:

```ts
document.getElementById(`omrs-icon-${icon}`) ? <Icon className={styles.icon} icon={`omrs-icon-${icon}`} /> : null;
```